pkg/apis/admissionregistration/validation/validation.go

```diff
@@ -194,28 +195,50 @@ func validateAdmissionReviewVersions(versions []string, requireRecognizedVersion
 	return allErrors
 }
 
-func ValidateValidatingWebhookConfiguration(e *admissionregistration.ValidatingWebhookConfiguration) field.ErrorList {
-	return validateValidatingWebhookConfiguration(e, true)
+// ValidateValidatingWebhookConfiguration validates a webhook before creation.
+func ValidateValidatingWebhookConfiguration(e *admissionregistration.ValidatingWebhookConfiguration, gv schema.GroupVersion) field.ErrorList {
+	// Require unique names when creating a webhook via any version other than admissionregistration.k8s.io/v1beta1
+	requireUniqueWebhookNames := gv != (schema.GroupVersion{Group: admissionregistration.GroupName, Version: "v1beta1"})
+
+	return validateValidatingWebhookConfiguration(e, true, requireUniqueWebhookNames)
 }
 
-func validateValidatingWebhookConfiguration(e *admissionregistration.ValidatingWebhookConfiguration, requireRecognizedVersion bool) field.ErrorList {
+func validateValidatingWebhookConfiguration(e *admissionregistration.ValidatingWebhookConfiguration, requireRecognizedVersion, requireUniqueWebhookNames bool) field.ErrorList {
 	allErrors := genericvalidation.ValidateObjectMeta(&e.ObjectMeta, false, genericvalidation.NameIsDNSSubdomain, field.NewPath("metadata"))
+	hookNames := sets.NewString()
 	for i, hook := range e.Webhooks {
 		allErrors = append(allErrors, validateValidatingWebhook(&hook, field.NewPath("webhooks").Index(i))...)
 		allErrors = append(allErrors, validateAdmissionReviewVersions(hook.AdmissionReviewVersions, requireRecognizedVersion, field.NewPath("webhooks").Index(i).Child("admissionReviewVersions"))...)
+		if requireUniqueWebhookNames && len(hook.Name) > 0 {
+			if hookNames.Has(hook.Name) {
+				allErrors = append(allErrors, field.Duplicate(field.NewPath("webhooks").Index(i).Child("name"), hook.Name))
```

Comment:
what's the reason we don't have a field path arguments in these validation funcs?

Reply:
ok, it's the root object. nvmd. Would be nil anyway.